src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -1045,4 +1050,41 @@ public static List<ClientTooltipComponent> gatherTooltipComponents(ItemStack sta
                 .toList();
     }
 
+    public static void onGatherStaticChunkGeometry(
+            ChunkBufferBuilderPack chunkBufferBuilderPack,
+            RenderChunkRegion region,
+            BlockPos.MutableBlockPos origin,
+            VisGraph visibilityGraph,
+            ChunkRenderDispatcher.CompiledChunk compiledChunk,
+            Consumer<BufferBuilder> beginLayer
+    ) {
+        MinecraftForge.EVENT_BUS.post(new GatherGeometryEvent.ChunkSectionStatic(
+                t -> {
+                    var buf = chunkBufferBuilderPack.builder(t);
+                    Preconditions.checkNotNull(buf, "The requested render type is not configured for static rendering: " + t);
```

Comment:
I disagree. From the NullPointerException Javadocs:
> Applications should throw instances of this class to indicate other illegal uses of the null object

Additionally, the JDK built-in null validation method `Objects.requireNonNull`, which "is designed primarily for doing parameter validation in methods and constructors" also throws `NullPointerException`

Reply:
Respectfully, I disagree with your disagreement. If the value being validated by `checkNotNull` is `t`, then it would be appropriate to throw `NullPointerException`, as the validation checks that the caller's provided value is not a `null` object.

However, the value under validation is `buf`, which is the result of `ChunkBufferBuilderPack#builder` with the `t` parameter passed in as an argument. This is why `NullPointerException` is inappropriate here: a caller which provides a non-`null` argument should not be expecting a `NullPointerException` thrown when their argument is somehow invalid (as they'd expect `IllegalArgumentException`). 

The details of how the `t` parameter is determined to be invalid is an implementation detail[^1] -- in this case, checking if the return value of `ChunkBufferBuilderPack#builder` for the given render type is `null`, which indicates that the render type is not configured for static rendering and therefore invalid. `NullPointerException` leaks this implementation detail to the caller.

[^1]: It is an implementation detail because the way by which the validity of the argument can change without affecting the contract of the method, such as invoking a helper method in another class which returns a `boolean` (and the way by which that method checks the validity of the argument is an implementation detail itself).

Reply:
I accept your disagreement :smile: